### ISSUE TITLE
MINSIGSTKSZ no longer a constant

### ISSUE THREAD

**Issue by btmcg** [catchorg/Catch2#2421]
**Describe the bug**
Building with glibc v2.35 causes a compiler error. Apparently MINSIGSTKSZ is no longer a constant in glibc.

```
In file included from /usr/include/signal.h:328,
                 from third_party/catch2/2.13.9/single_include/catch2/catch.hpp:8030,
                 from /home/brian/src/repos/protocol-parser/test/main.cpp:3:
third_party/catch2/2.13.9/single_include/catch2/catch.hpp:10818:58: error: call to non-‘constexpr’ function ‘long int sysconf(int)’
10818 |     static constexpr std::size_t sigStackSize = 32768 >= MINSIGSTKSZ ? 32768 : MINSIGSTKSZ;
      |                                                          ^~~~~~~~~~~
In file included from /usr/include/c++/11.2.0/bits/atomic_wait.h:44,
                 from /usr/include/c++/11.2.0/bits/atomic_base.h:41,
                 from /usr/include/c++/11.2.0/bits/shared_ptr_atomic.h:33,
                 from /usr/include/c++/11.2.0/memory:78,
                 from third_party/catch2/2.13.9/single_include/catch2/catch.hpp:2943,
                 from /home/brian/src/repos/protocol-parser/test/main.cpp:3:
/usr/include/unistd.h:640:17: note: ‘long int sysconf(int)’ declared here
  640 | extern long int sysconf (int __name) __THROW;
      |                 ^~~~~~~
In file included from /home/brian/src/repos/protocol-parser/test/main.cpp:3:
third_party/catch2/2.13.9/single_include/catch2/catch.hpp:10877:45: error: size of array ‘altStackMem’ is not an integral constant-expression
10877 |     char FatalConditionHandler::altStackMem[sigStackSize] = {};
```

This can be fixed by adding `#define CATCH_CONFIG_NO_POSIX_SIGNALS` before the catch include.

**Expected behavior**
Linux should be able to compile this without issue.

**Reproduction steps**
Use glibc 2.35.

**Platform information:**
 - OS: Arch Linux
 - Compiler+version: gcc-11.2.0

 - Catch version: **v2.13.9

**Additional context**
https://stackoverflow.com/questions/71454588/minsigstksz-error-after-update-in-my-manjaro-linux
https://sourceware.org/git/?p=glibc.git;a=blob;f=NEWS;h=85e84fe53699fe9e392edffa993612ce08b2954a;hb=HEAD

**Comment by horenmar** [comment]
Your catch.hpp is old, this is 2.13.9 at the indicated line: https://github.com/catchorg/Catch2/blob/d71b4617e9935f8589870af211f5b7552d743654/single_include/catch2/catch.hpp#L10818

In fact the version that fixed this issue was 2.13.5: https://github.com/catchorg/Catch2/blob/devel/docs/release-notes.md#2135

**Comment by Serge3leo** [comment]
@davidmcgrew , thanks. You workaround (define `CATCH_CONFIG_NO_POSIX_SIGNALS`) work fine for v2.4.2 single_include/catch2.

But, last version v2.13.10 single_include/catch2 (for example, included in Debian 12.8) work fine by default.
